src/Orchard/Tasks/Locking/Migrations/FrameworkMigrations.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Orchard.Data.Migration;
+
+namespace Orchard.Tasks.Locking.Migrations {
+    public class FrameworkMigrations : DataMigrationImpl {
+
+        public int Create() {
+            SchemaBuilder.CreateTable("DistributedLockRecord", table => table
+                .Column<int>("Id", column => column.PrimaryKey().Identity())
+                .Column<string>("Name", column => column.NotNull().WithLength(256))
+                .Column<string>("MachineName", column => column.WithLength(256))
+                .Column<int>("ThreadId", column => column.Nullable())
+                .Column<int>("Count")
+                .Column<DateTime>("CreatedUtc")
+                .Column<DateTime>("ValidUntilUtc"));
```

Comment:
Good question, I don't know. On the one hand, it would allow even greater control over the duration of a lock, but at the same time it would be risky when acquired for a thread that is relatively short-lived (as opposed to a machine name, in which case it's not as devastating if code fails to release the lock, since the locking code is re-entrant). On the other hand, anyone can shoot themselves in the foot by using a far away date anyway, and if the intention is to have an eternal lock until explicitly released, then it may be nicer to be explicit about it by passing in NULL for `ValidUntilUtc`. If you agree, I'll change.